### ISSUE TITLE
Bookmarks bar folder and sub-folders slow to open.

### ISSUE THREAD

**Issue by rereser** [IDA-RE-things/Chrome-xp-api-adapter#26]
OS: Win XP SP3
Browser: Supermium 126.0.6478.254 R4
chrome-xpapi-adapter.1.2.0.5080.beta.dll renamed to chrome-xpapi-adapter.dll.

I have a folder in the bookmarks bar with subfolders that i use to access my bookmarks.
Opening of sub-folders "on mouse hover" now takes 1 to 3 seconds.
The same behavior on single bookmark icons without text to display the url.
No such issues with v5078.

On a positive note, issue #25 is resolved.

**Comment by IDA-RE-things** [comment]
I have uploaded 5080.beta2.dll 
Is it working for you? I have tested it with folder on bookmarks bar with many links inside.

At all its a side effect of my work and tuning the browser _to consume minimum of CPU_.
I will continue to experiment, if this works not so fine on another configurations.

Also if you can, test it with a previous Sm build v126-r3, which has maxumum optimization/tuning curretly with my DLL.

**Comment by rereser** [comment]
Issue still there with the chrome-xpapi-adapter.1.2.0.5080.beta2.dll.
It is better but not resolved completely.
Still a delay on some subfolders and single bookmarks.
It happens also on subfolders with a single bookmark so not related to the amount of bookmarks in a folder.

**Comment by rereser** [comment]
Just tested on 126 R3, same issue, slight delay on random subfolders and single bookmarks.
But you are definitively on to something here!
https://browserbench.org/Speedometer2.1
When running the test in an incognito window the results here went from 145 (v5078.dll) to 164 (5080 beta1.dll) to 191 (5080 beta2.dll)
This test was done on 126 R4, on 126 R3 the incognito window seems broken with 5080 B1 and 5080 B2.
Edit: same test results on 126 R3, but the incognito window is not dark as expected.
Don't give credits to those tests normally but the difference is too much to ignore.

**Comment by rereser** [comment]
Thanks for the chrome-xpapi-adapter.1.2.0.5080.beta3.dll.
No measurable improvement over the beta2.dll with the bookmarks folder delay on mouse hover.
Have been using the beta2.dll since you released and it is far better than the beta1.dll
Delays on random folders, when it occurs, are now under a second.
But the browserbench test here went from 195 to around 160 with the beta3.dll.
Tested both 126 R3 and R4.

**Comment by IDA-RE-things** [comment]
In the latest today beta4 upload , the delay issue was fixed (appropriate code causing this was disabled). And D3DCompiler was updated (included to archive, not forgot to update also).
Also I have added small optimizations, to additionally minimize overal CPU load.
My tests with MotionMark show me ~same results as original distrib. And sometime slightly faster, when HW acceleration disabled on both orig and my. (SW rendering only used)
